### PR TITLE
Add "Assault Move" modifier to attack move

### LINKED ISSUE #14008: Toggle/switch for "attack non-combat buildings"
A lot of words has already been said on it, so I just want to bring it to the github plane.

Looks like a new A-move modifier will be introduced in the next playtest (ctrl a-move), allowing the old attack behaviour when army attacks all units and buildings. [here](http://www.sleipnirstuff.com/forum/viewtopic.php?f=82&p=304663#304663) @pchote mentions that it's quite a lot of work to make it a toggle rather than modifier.

Still, we think this feature should be considered to be made as a toggle in the future, even if not for the upcoming release. All the details on the suggestion can be read [here](http://www.sleipnirstuff.com/forum/viewtopic.php?f=82&p=303590#303590).

### PR DESCRIPTION
**Pull request by pchote**
As discussed in other tickets, IRC, and on the forum.  This follows the lead from C&C3 in providing an a quick and stateless attack-anything + attack move variation.

This also fixes autotargeting vs buildings in TS, and adds an attack-move cursor to d2k (in addition to assault-move cursors for RA and TD).  The second commit will need to be edited to work with the older hotkey setup when picking to prep.

Closes #14008

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/Ingame/CommandBarLogic.cs
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/Ingame/CommandBarLogic.cs]: Note: this will be fixed by the next phase of my hotkey rework, but that is currently blocked on #13781, #13966, #13974 and in any case won't help on the prep branch.

File: OpenRA.Mods.Common/Traits/AttackMove.cs
Comment (obrakmann) [inline OpenRA.Mods.Common/Traits/AttackMove.cs]: ```OpenRA.Mods.Common/Traits/AttackMove.cs:L121: [SA1306] Variable names and private field names must start with a lower-case letter: ExpectedButton.```

**Review by obrakmann — COMMENTED** [review]
Looks good and works as expected.

I didn't see any new cursor in TS, is that intentional?

**Comment by pchote** [comment]
> I didn't see any new cursor in TS, is that intentional?

Yeah, we don't have any attack move cursors in TS yet, and I don't have a functional indexed-image editor to make them.  @reaperrr graciously did ra/cnc/d2k cursors here, and I didn't want to push my luck asking for another two for ts.

**Comment by pchote** [comment]
Fixed.

**Review by obrakmann — DISMISSED** [review]
👍 

The way this works doesn't stand in the way of the attack-move rewrite, either. The condition granting/revoking can then eventually be moved from `AttackMove.Tick` to `AttackMoveActivity.On(First|Last)Run`.

**Comment by pchote** [comment]
Yup, that was my plan/hope.  The conditions should then also hopefully stay active all the time, instead of being dropped when the actor is attacking, which will make them useful for more than just toggling target types.

**Comment by pchote** [comment]
Added cursors for TS again thanks to @reaperrr.

**Comment by reaperrr** [comment]
Looks good to me and since only the TS cursors changed I consider @obrakmann's +1 still valid, but before I merge this: @pchote would you mind doing the picking once merged, since I'm not sure how much the 2nd commit needs to be edited for prep? I don't want to mess that up.

**Comment by pchote** [comment]
Yup, can do, although it will need to be tomorrow.

**Comment by pchote** [comment]
I pushed an [assault-move-playtest](https://github.com/pchote/OpenRA/tree/assault-move-playtest) branch that is based on prep-1707.  You can cherry pick safely from that when merging.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/5cb4ecfdfe59af62870aba3027294482b095041b...745f15c839898d3bce7748169ceb6fc13f088056

File: OpenRA.Mods.Common/Traits/AttackMove.cs
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/AttackMove.cs]: Why `generic-blocked` instead of `assaultmove-blocked`?

Comment (pchote) [reply OpenRA.Mods.Common/Traits/AttackMove.cs]: This was maintaining the existing behaviour - we didn't have any assaultmove-blocked cursors at the time.

**Pull request merged**
